src/library.js

```diff
@@ -428,28 +428,37 @@ LibraryManager.library = {
     ___setErrNo(ERRNO_CODES.EINVAL);
     return -1;
   },
-  sbrk: function(bytes) {
+
+  // Implement a Linux-like 'memory area' for our 'process'.
+  // Changes the size of the memory area by |bytes|; returns the
+  // address of the previous top ('break') of the memory area
+  // We control the "dynamic" memory - DYNAMIC_BASE to DYNAMICTOP
+  sbrk__asm: true,
+  sbrk__sig: ['ii'],
+  sbrk: function(increment) {
+    increment = increment|0;
+    var dynamicTop = 0;
 #if USE_PTHREADS
-    if (ENVIRONMENT_IS_PTHREAD) return _emscripten_sync_run_in_main_thread_1({{{ cDefine('EM_PROXIED_SBRK') }}}, bytes);
+    dynamicTop = Atomics_add(HEAPU32, DYNAMICTOP_PTR>>2, increment);
 #endif
-    // Implement a Linux-like 'memory area' for our 'process'.
-    // Changes the size of the memory area by |bytes|; returns the
-    // address of the previous top ('break') of the memory area
-    // We control the "dynamic" memory - DYNAMIC_BASE to DYNAMICTOP
-    var self = _sbrk;
-    if (!self.called) {
-      DYNAMICTOP = alignMemoryPage(DYNAMICTOP); // make sure we start out aligned
-      self.called = true;
-      assert(Runtime.dynamicAlloc);
-      self.alloc = Runtime.dynamicAlloc;
-      Runtime.dynamicAlloc = function() { abort('cannot dynamically allocate, sbrk now has control') };
-    }
-    var ret = DYNAMICTOP;
-    if (bytes != 0) {
-      var success = self.alloc(bytes);
-      if (!success) return -1 >>> 0; // sbrk failure code
-    }
-    return ret;  // Previous break location.
+    dynamicTop = (dynamicTop + increment)|0;
+    dynamicTop = ((dynamicTop + 15) & -16)|0;
+#if !USE_PTHREADS
+    HEAPU32[DYNAMICTOP_PTR>>2] = dynamicTop;
+#endif
+    return dynamicTop|0; // TODO: error checking, return -1 on error and set errno.
```

Comment:
i believe this TODO must be done before landing, for memory growth and memory growth failure tests to pass.